ts/src/ellipx.ts

```diff
@@ -0,0 +1,1583 @@
+        if (dataArray.length > 0) {
+            const firstItem = dataArray[0];
+            const balanceDate = this.safeValue (firstItem, 'Balance_Date', {});
+            result['timestamp'] = this.safeInteger (balanceDate, 'unixms');
+            result['datetime'] = this.iso8601 (result['timestamp']);
+        }
+        // Process each balance entry
+        for (let i = 0; i < dataArray.length; i++) {
+            const entry = dataArray[i];
+            const balance = this.safeValue (entry, 'Balance', {});
+            const currency = this.safeString (balance, 'currency');
+            if (currency !== undefined) {
+                const account = {
+                    'free': this.parseAmount (entry['Unencumbered_Balance']['value_xint']),
+                    'used': this.parseAmount (entry['Liabilities']['value_xint']),
+                    'total': this.parseAmount (balance['value_xint']),
+                };
+                result[currency] = account;
+            }
+        }
+        return this.safeBalance (result);
+    }
+
+    /**
+     * @method
+     * @name ellipx#createOrder
+     * @description create a new order in a market
+     * @param {string} symbol unified market symbol (e.g. 'BTC/USDT')
+     * @param {string} type order type - the exchange automatically sets type to 'limit' if price defined, 'market' if undefined
+     * @param {string} side 'buy' or 'sell'
+     * @param {float} amount amount of base currency to trade (can be undefined if using Spend_Limit)
+     * @param {float|undefined} price price per unit of base currency for limit orders
+     * @param {object} [params] extra parameters specific to the EllipX API endpoint
+     * @param {float} [params.Spend_Limit] maximum amount to spend in quote currency (required for market orders if amount undefined)
+     * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+     * @throws {ArgumentsRequired} if neither amount nor Spend_Limit is provided
+     */
+    async createOrder (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        // the exchange automatically sets the type to 'limit' if the price is defined and to 'market' if it is not
+        const marketSymbol = market['symbol'].replace ('/', '_'); // Convert BTC/USDC to BTC_USDC
+        let Type = 'bid';
+        if (side === 'buy') {
+            Type = 'bid';
+        } else {
+            Type = 'ask';
+        }
+        const request: any = {
+            'currencyPair': marketSymbol,
+            'Type': Type,
+        };
+        if (amount === undefined && params['Spend_Limit'] === undefined) {
+            throw new ArgumentsRequired (this.id + ' createOrder requires an amount or Spend_Limit');
+        }
+        if (amount !== undefined) {
+            request['Amount'] = amount.toString ();
```

Comment:
we should not do number.toString(), ideally we should sue `amounttoPrecision` and if he precision is not available then we can use `numberToString()`

Reply:
fix to `amountToPrecision`.